### PR TITLE
DTSPO-5507 - add smoke tests

### PR DESCRIPTION
**Pull request by louisehuyton**
### JIRA link (if applicable) ###
DTSPO-5507

### Change description ###
Added smoke tests

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/smoke/smoke.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { fail } from 'assert';
+import axios, { AxiosResponse } from 'axios';
+
+jest.retryTimes(20); // 20 retries at 1 second intervals
+jest.setTimeout(15000);
+
```
Comment (timja) [inline src/test/smoke/smoke.ts:6]:
```suggestion
const testUrl = process.env.TEST_URL || 'https://localhost:8080'
```

File: src/test/smoke/smoke.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+const servicesToCheck = [
+  { name: 'Default Page Template', url: 'http://localhost:8080', heading: '<h1 class="govuk-heading-xl">Default page template</h1>', },
+  //add more services here when created
+];
```
Comment (timja) [inline src/test/smoke/smoke.ts:10]: remove this

**Review by msl8r — APPROVED** [review]

File: src/test/smoke/smoke.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+describe('Smoke Test', () => {
+  describe('Health Check', () => {
```
Comment (timja) [inline src/test/smoke/smoke.ts:13]: remove this health check test it's meaningless

File: src/test/smoke/smoke.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { fail } from 'assert';
+import axios, { AxiosResponse } from 'axios';
+
+jest.retryTimes(20); // 20 retries at 1 second intervals
+jest.setTimeout(15000);
+
+const servicesToCheck = [
+  { name: 'Default Page Template', url: 'http://localhost:8080', heading: '<h1 class="govuk-heading-xl">Default page template</h1>', },
```
Comment (timja) [inline src/test/smoke/smoke.ts:8]: best just to inline this and remove the .each, i.e. just check the home page

File: src/test/smoke/smoke.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        await expect(checkService()).resolves.not.toThrow();
+      });
+    });
+  });
+
+  describe('UI Test', () => {
+    describe.each(servicesToCheck)('Required services should have a heading present with the correct text displayed', ({ name, url, heading }) => {
+      test(`${name}`, async () => { // eslint-disable-line @typescript-eslint/no-empty-function
+        try {
+          const response: AxiosResponse = await axios.get(url, {
```
Comment (timja) [inline src/test/smoke/smoke.ts:43]:
```suggestion
          const response: AxiosResponse = await axios.get(testUrl, {
```

**Review by timja — CHANGES_REQUESTED** [review]

File: src/test/smoke/smoke.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { fail } from 'assert';
+import axios, { AxiosResponse } from 'axios';
+
+jest.retryTimes(20); // 20 retries at 1 second intervals
+jest.setTimeout(15000);
```
Comment (timja) [inline src/test/smoke/smoke.ts:5]: I would start without this
```suggestion
```

File: src/test/smoke/smoke.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+          } catch (e) {
+            await new Promise((resolve, reject) =>
+              setTimeout(() => reject(`Endpoint is not up: '${parsedUrl}': ${e}`), 1000)
+            );
+          }
+        };
+        await expect(checkService()).resolves.not.toThrow();
+      });
+    });
+  });
```
Comment (timja) [inline src/test/smoke/smoke.ts:32]: can you remove this please, we shouldn't be testing htis
```suggestion
```

File: src/test/smoke/smoke.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+  describe('UI Test', () => {
```
Comment (timja) [inline src/test/smoke/smoke.ts:34]:
```suggestion
  describe('Home page loads with correct content', () => {
```

File: src/test/smoke/smoke.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    describe('Required services should have a heading present with the correct text displayed', () => {
```
Comment (timja) [inline src/test/smoke/smoke.ts:35]:
```suggestion
```

File: src/test/smoke/smoke.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      test(`${testUrl}`, async () => { // eslint-disable-line @typescript-eslint/no-empty-function
```
Comment (timja) [inline src/test/smoke/smoke.ts:36]:
```suggestion
      test('with correct content', async () => {
```

File: src/test/smoke/smoke.ts [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        try {
+          const response: AxiosResponse = await axios.get(testUrl, {
+            headers: {
+              'Accept-Encoding': 'gzip',
+              accept: 'application/json',
+            },
+          });
+          if (!response.data.includes('<h1 class="govuk-heading-xl">Default page template</h1>')) {
+            console.log(response.data)
+            throw new Error(` Heading not present and/or correct`)
```
Comment (timja) [inline src/test/smoke/smoke.ts:46]:
```suggestion
            throw new Error('Heading not present and/or correct')
```

**Review by timja — COMMENTED** [review]
something like this

File: src/test/smoke/smoke.ts [hunk 12]
```diff
@@ -0,0 +1,24 @@
+import { fail } from 'assert';
+import axios, { AxiosResponse } from 'axios';
+
+const testUrl = process.env.TEST_URL || 'http://localhost:8080'
+
+describe('Smoke Test', () => {
+  describe('Home page loads with correct content', () => {
```
Comment (timja) [inline src/test/smoke/smoke.ts:7]:
```suggestion
  describe('Home page loads', () => {
```

**Review by timja — APPROVED** [review]
if it passes

File: Jenkinsfile_CNP [hunk 13]
```diff
@@ -13,4 +13,13 @@ withPipeline(type, product, component) {
...
   }
 
+  after('Smoke Test - AKS sandbox') {
+    steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'smoke-output/**/*'
+  }
+
+  after('Functional Test - AKS sandbox') {
+    steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'functional-output/***/**/*'
+    steps.junit allowEmptyResults: true, testResults: './functional-output/***/**/*-testsuite.xml'
+  }
```
Comment (timja) [inline Jenkinsfile_CNP:23]: you shouldn't be adding these, the app should be configured to output to the right location

**Pull request merged**